iceoryx_posh/test/moduletests/test_shared_memory_user.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
```

Comment:
We are very thankful for every contribution but we require profound C++ knowledge as we stated before multiple times in our developer meetups. We also require that you please follow our coding guidelines.

- If you have architectural questions we are happy to help and provide you with the correct documentation.
- If you have a lack of C++ knowledge we can provide you with some book recommendations (see the last section of  https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md) so that you can refresh your knowledge before starting to contribute.

The reason for those requirements are that we develop safety critical software where maybe at some time in the future human lifes will depend on and we all want to make sure that we deliver the best possible software.

In this pull request we see a basic lack of knowledge of very basic C and C++ concepts. Furthermore, it seems that the basic functionality of iceoryx is not fully understood and the tests are actually testing nothing.

Reply:
@elfenpiff I only see comments for the file header. What are the code parts that led to your conclusions? I think it would be fair to talk about specific things that do not follow the contribution guidelines or are wrong.

Reply:
@budrus You are completely right, this is fair!

@manel94 sorry for that.

Reply:
Thanks @elfenpiff